contentgrid-spring-data-rest/src/main/java/com/contentgrid/spring/data/rest/webmvc/blueprint/TitleMessageSourceResolvable.java

```diff
@@ -0,0 +1,39 @@
+package com.contentgrid.spring.data.rest.webmvc.blueprint;
+
+import com.contentgrid.spring.data.rest.mapping.Property;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.Value;
+import org.springframework.context.MessageSourceResolvable;
+import org.springframework.data.rest.webmvc.RootResourceInformation;
+
+@Value
+public class TitleMessageSourceResolvable implements MessageSourceResolvable {
+    String[] codes;
+
+    public static TitleMessageSourceResolvable forEntity(RootResourceInformation information) {
+        return forNestedProperty(information, List.of());
+    }
+
+    public static TitleMessageSourceResolvable forProperty(RootResourceInformation information, Property property) {
+        return forNestedProperty(information, List.of(property));
+    }
+
+    public static TitleMessageSourceResolvable forNestedProperty(RootResourceInformation information, Collection<Property> properties) {
```

Comment:
To add titles to a nested property, you need to use `com.contentgrid.spring.data.support.auditing.v1.AuditMetadata.createdDate._title` as key if you want to see them in json schema. `com.contentgrid.userapps.demo.test.model.Invoice.auditMetadata.createdDate._title` doesn't work.